ci/kokoro/docker/build.sh

```diff
@@ -506,6 +506,13 @@ docker_flags=(
   "--rm"
 )
 
+# Some machines have complex host network configurations that we want to shared
+# with the running docker container. Setting this environment variable tells
+# the docker container to use the host network rather than a bridge network.
+if [[ -n "${DOCKER_USE_HOST_NETWORK:-}" ]]; then
+  docker_flags+=(--net host)
```

Comment:
It doesn't seem to do it for me.  That is, the diagnosis test case I had set up still fails.

If I wait for a while (presumably to clear the throttling cache), the first error occurs at the same point.  There is a difference after that however, as it seems to timeout earlier, which leads to additional (faster) failures.

Reply:
Hmm. I don't understand why you'd still be seeing DNS flakiness since using the host network should mean the docker container is now using your host's DNS cache. Anyway, I dropped this PR in favor of https://github.com/googleapis/google-cloud-cpp/pull/4288, which seems to address the issues I was seeing (which I thought were the same as the issues you were seeing). Maybe try that new PR and then let me know exactly what errors you're seeing.